From ad387f9035a3edee0de95255e26b28dd440f2fb5 Mon Sep 17 00:00:00 2001
From: Prabhu Sundararaj <prabhu.sundararaj@nxp.com>
Date: Mon, 24 Apr 2017 14:47:25 -0500
Subject: [PATCH 1821/5242] ARM64: DMA: Export dma operation APIs for arm64
 platform

commit  8c254766ba571678ec9fe2f84b6f07f0a2c9172b from
https://source.codeaurora.org/external/imx/linux-imx.git

Some DMA operation APIs are not exported, so when load driver as module,
there will error saying some APIs can not be found.
This patch exported these DMA related APIs.

Signed-off-by: Shawn Xiao <b49994@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/mm/dma-mapping.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm64/mm/dma-mapping.c b/arch/arm64/mm/dma-mapping.c
index 935c787..a5450e2 100644
--- a/arch/arm64/mm/dma-mapping.c
+++ b/arch/arm64/mm/dma-mapping.c
@@ -32,6 +32,10 @@
 
 #include <asm/cacheflush.h>
 
+EXPORT_SYMBOL(__dma_map_area);
+EXPORT_SYMBOL(__dma_unmap_area);
+EXPORT_SYMBOL(__dma_flush_area);
+
 static int swiotlb __ro_after_init;
 
 static pgprot_t __get_dma_pgprot(unsigned long attrs, pgprot_t prot,
-- 
1.7.9.5

